pkg/kubefed2/federate/util.go

```diff
@@ -101,29 +103,37 @@ func LookupAPIResource(config *rest.Config, key, targetVersion string) (*metav1.
 			continue
 		}
 		for _, resource := range resourceList.APIResources {
+			var matchResource = func() {
```

Comment:
Nice cleanup!  In the interest of minimizing indentation and simplifying loop construction, consider moving the function definition outside of the loop and accept parameters for the loop-local variables, e.g.

```golang
var matchResource = func(resource metav1.APIResource, gv schema.GroupVersion) {
...
}
```